docs/advanced-topics/session-management.md

```diff
@@ -0,0 +1,26 @@
+
+When we have such a way to uniquely identify a player, we can then use it to map each player's data that would be needed when reconnecting, such as the current state of their character for example. Then, we need to make sure this data is up to date and to keep it after a player disconnects. We could add a timeout after which we can purge the data or we could decide to clear all data only when a session completes. And finally, once we have up to date data that we can link to a specific player who disconnected earlier, we can now handle reconnection properly.
+
+## Handling reconnection
+In cases where we use a Player Object, a new Default Player Prefab will be automatically spawned when any player connects to the game (including if they are reconnecting). We can then use the player's data to update that object so that its state returns to what it was before disconnecting. In those cases, we would need to keep all the important data that we want to restore and map it to the player using our identification system.
+
+In cases where we use the ownership model, we could instead keep the objects that a player owns when they disconnects, and set the reconnected player as their new owner. To accomplish this, the only data we would need to keep would be the mapping between those objects and their owning player's identifier, then when a player reconnects we could use this mapping to set them as the new owner.
+
+## Example using the Player Object model
+Here is an example from the Boss Room sample, showing some simple session management. The game uses the Player Object model and a GUID to identify unique players. In that example, each player's data is saved in the `m_ClientData` dictionary, and each of their clientId is mapped to their Guid in `m_ClientIDToGuid`. The saved data is represented by the generic type `T`, which is a struct implementing the `ISessionPlayerData` interface. This struct contains all the data required. An implementation could contain, for example, the character's current hit points, position, class, etc. The interface has two properties, `IsConnected` and `ClientID`, and the method `Reinitialize`. `IsConnected` is used in the `OnClientApprovalCheck` method to differentiate between a duplicate connection and a reconnection, and is set when a player connects to the game and is unset when they disconnect. `ClientID` is used to link the player's session data with their current ClientID. It is updated when they connect or reconnect. Finally, `Reinitialize` is called in the `OnSessionEnded` method for each player's session data. This should reset the data of the player's session to represent a new session.
```

Comment:
Do we really need to describe the code we wrote? The code is pretty self describing.

Reply:
The "Why" is usually the most important part

Reply:
is the code self describing to everyone or just if you know what it does...is this something that is useful if it is then let's leave it in place

Reply:
This page is for programmers, if they can read code, they might not need chunks of this doc (what do you think @LPLafontaineB ?)
It might be worth it to talk about how boss room uses it, more than what the session manager itself does (as it's designed to be reused)

Reply:
Rereading it, I think there is definitely part of it that's not useful and could be cut, while adding something explaining the reasons for our choices and how it is used inside BossRoom a bit better

Reply:
I am going to ask the question again, does leaving the extra in hinder an experienced coder as much as it may help someone learning? that's the question. Remember that while the target audience may be experienced its not a bad thing if we also help the others.

Reply:
In this case, it's more that I was repeating myself. This extra info was describing something that is already explained by the comments in the code itself, or that is self-describing code. I don't think removing those parts would hinder someone learning.

Reply:
Although I did have my nose inside this system for weeks, so there might be some parts of it that seem evident to me but are not

Reply:
A bigger conversation could be about how to address experienced coders vs beginners. Could we have a "beginner explanation" collapsible section? Something that would allow experienced devs to skip to the part they want?
Or have a TLDR section at the top with the essentials for experienced devs and then the rest of the articles for beginners? It's always an issue I have with some of these docs, is being able to get to the information I really need without getting the whole world explained to me yet again.

Reply:
For this specific PR however, should we wait for this conversation's conclusion? In this case I feel that the extra information I removed is mostly clutter and doesn't really help to explain the topic, so either way I think it should be fine on that front, but I'm open to add more details if we prefer going that way instead.

Reply:
Very old article that kept popping up in my last 7 years of career creating and documenting code:
https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/
Code tells you how, comments tell you why. This doc should be telling you the why and point to the code for the how. You don't want to reimplement our code in plain text, that's useless, gets out of date very quickly and our users don't need this (even non-programmers, as if they don't understand code, they won't understand code-in-plain-text). It's just adding easily outdated noise. Doc should be as straightforward as possible, else readers will just glance over it and miss the important bits.
If you think this adds to your code, if there's a trick you used that could be worth explaining, then sure (for example anything related to managing GUIDs and the discussions we've had in the auth channel). But implementation details should be skipped. Users could want to implement this their own way as well, they don't HAVE to implement it the way you did.

Reply:
Makes sense. In that case, I think we should keep it how it currently is